### PR TITLE
fix(pnp): use original _findPath when isMain is true

### PR DESCRIPTION
**Pull request by merceyz**
**What's the problem this PR addresses?**

Node is unable to detect if it should run under ESM or not when the PnP hook is registered because it returns `false` in `Module._findPath` which causes this check to be false
https://github.com/nodejs/node/blob/22c52aac6ace9154e9424c2994289ee4dde12bab/lib/internal/modules/run_main.js#L66-L68

**How did you fix it?**

Call the original `Module._findPath` when `isMain` is true

**Checklist**
- [x] I have read the [Contributing Guide](https://yarnpkg.com/advanced/contributing).
- [x] I have set the packages that need to be released for my changes to be effective.
- [x] I will check that all automated PR checks pass before the PR gets reviewed.

### CODE REVIEW

**Comment by arcanis** [comment]
I'm not sure that works - running `node lodash` should resolve `lodash` as bare identifier and execute it, but with this change the resolution will fail 🤔 What's the exact problem that makes this function return `false`? That `.mjs` extensions are ignored during the resolution?

**Comment by merceyz** [comment]
> I'm not sure that works - running node lodash should resolve lodash as bare identifier and execute it, but with this change the resolution will fail thinking

The path seems to be absolute when it comes from node so can check that as well

> What's the exact problem that makes this function return false? That .mjs extensions are ignored during the resolution?

`paths` is null

File: packages/yarnpkg-pnp/sources/loader/applyPatch.ts [hunk 1]
```diff
@@ -320,7 +320,12 @@ export function applyPatch(pnpapi: PnpApi, opts: ApplyPatchOptions) {
     if (request === `pnpapi`)
       return false;
 
-    if (!enableNativeHooks)
+    // Node sometimes call this function with an absolute path and a `null` set
+    // of paths. This would cause the resolution to fail. To avoid that, we
+    // fallback on the regular resolution. We only do this when `isMain` is
+    // true because the Node default resolution doesn't handle well in-zip
+    // paths, even absolute, so we try to use it as little as possible.
+    if (!enableNativeHooks || (isMain && npath.isAbsolute(request)))
```
Comment (arcanis) [inline packages/yarnpkg-pnp/sources/loader/applyPatch.ts:328]: @merceyz can you check this comment is accurate?

Comment (merceyz) [reply packages/yarnpkg-pnp/sources/loader/applyPatch.ts:328]: Yup, that's exactly what happens :+1:

**Pull request merged**